specutils/spectra/spectrum1d.py

```diff
@@ -196,33 +239,52 @@ def __init__(self, flux=None, spectral_axis=None, wcs=None,
                 if len(temp_axes) != 1:
                     raise ValueError("Input WCS must have exactly one axis with "
                                      "spectral units, found {}".format(len(temp_axes)))
+                else:
+                    # Due to FITS conventions, the WCS axes are listed in opposite
+                    # order compared to the data array.
+                    self._spectral_axis_index = len(flux.shape)-temp_axes[0]-1
+
+                if move_spectral_axis is not None:
+                    if move_spectral_axis.lower() == 'first':
+                        move_to_index = 0
+                    elif move_spectral_axis.lower() == 'last':
+                        move_to_index = wcs.naxis - 1
+                    else:
+                        raise ValueError("move_spectral_axis must be either 'first' or 'last'")
+
+                    if move_to_index != self._spectral_axis_index:
+                        wcs = wcs.swapaxes(self._spectral_axis_index, move_to_index)
+                        if flux is not None:
+                            flux = np.swapaxes(flux, self._spectral_axis_index, move_to_index)
+                        if "mask" in kwargs:
+                            if kwargs["mask"] is not None:
+                                kwargs["mask"] = np.swapaxes(kwargs["mask"],
```

Comment:
Just a note on this - I think it's fine as is, but we may need to add a convenience tool to reorder the celestial dimensions.  If you give a cube with dims [RA, Dec, Spectral] and move_to_first now, you'll get [Spectral, Dec, RA], which is not what anyone would normally expect (imo).  I don't think we should handle the logic here, but we might push a function in astropy.wcs that would `rearrange_celestial_to_have_longitude_first`

Reply:
spectral-cube does this:
https://github.com/radio-astro-tools/spectral-cube/blob/master/spectral_cube/cube_utils.py#L204

Reply:
Agreed - FYI this is currently what specutils does by default, so it's not any worse than it is currently 🙈 .

Reply:
I added a note to the docs for now saying that RA/Dec might not be in a sensible order if you move the spectral axis.